imap/http_jmap.c

```diff
@@ -1052,11 +1057,12 @@ static int jmap_upload(struct transaction_t *txn)
     const char *type = "application/octet-stream";
     if ((hdr = spool_getheader(hdrcache, "Content-Type"))) {
         type = hdr[0];
+        /* Remove CFWS and encodings from type */
+        normalisedtype = charset_decode_mimeheader(type, CHARSET_KEEPCASE);
```

Comment:
This means "type" in the response object will now be `null` instead of `application/octet-stream` if one isn't provided during upload? Is that expected?

Reply:
That's an error. I updated the code and added a Cassandane test.